data/keystore/src/test/java/tech/pegasys/artemis/bls/keystore/KeyStoreTest.java

```diff
@@ -0,0 +1,262 @@
+      final Path pbkdf2TestVectorPath =
+          Paths.get(Resources.getResource(unsupportedChecksumFunctionJsonResource).toURI());
+      return Stream.of(Arguments.of(scryptTestVectorPath), Arguments.of(pbkdf2TestVectorPath));
+    } catch (URISyntaxException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @SuppressWarnings("UnusedMethod")
+  private static Stream<Arguments> kdfParams() {
+    // KdfParam, expected checksum, expected encrypted cipher message
+    return Stream.of(
+        Arguments.of(
+            SCryptParamBuilder.aSCryptParam().withSalt(SALT).build(),
+            Bytes.fromHexString("149aafa27b041f3523c53d7acba1905fa6b1c90f9fef137568101f44b531a3cb"),
+            Bytes.fromHexString(
+                "54ecc8863c0550351eee5720f3be6a5d4a016025aa91cd6436cfec938d6a8d30")),
+        Arguments.of(
+            Pbkdf2ParamBuilder.aPbkdf2Param().withSalt(SALT).build(),
+            Bytes.fromHexString("18b148af8e52920318084560fd766f9d09587b4915258dec0676cba5b0da09d8"),
+            Bytes.fromHexString(
+                "a9249e0ca7315836356e4c7440361ff22b9fe71e2e2ed34fc1eb03976924ed48")));
+  }
+
+  @SuppressWarnings("UnusedMethod")
+  private static Stream<Arguments> basicKdfParam() {
+    return Stream.of(
+        Arguments.of(
+            SCryptParamBuilder.aSCryptParam().build(), CipherParamBuilder.aCipherParam().build()),
+        Arguments.of(
+            Pbkdf2ParamBuilder.aPbkdf2Param().build(), CipherParamBuilder.aCipherParam().build()));
+  }
+
+  @ParameterizedTest
+  @MethodSource("keystorePaths")
+  void loadKeyStoreAndDecryptKey(final Path keyStorePath) throws Exception {
+    final KeyStore keyStore = KeyStoreLoader.loadFromFile(keyStorePath);
+    final KeyStoreData keyStoreData = keyStore.getKeyStoreData();
+    assertNotNull(keyStoreData);
+    final KdfParam param = keyStoreData.getCrypto().getKdf().getParam();
+    assertNotNull(param);
+    if (keyStoreData.getCrypto().getKdf().getCryptoFunction() == SCRYPT) {
+      assertTrue(param instanceof SCryptParam);
+    } else if (keyStoreData.getCrypto().getKdf().getCryptoFunction() == PBKDF2) {
+      assertTrue(param instanceof Pbkdf2Param);
+    } else {
+      fail("Unsupported crypto function");
+    }
+    assertTrue(keyStore.validatePassword(PASSWORD));
+    assertFalse(keyStore.validatePassword("test"));
+
+    final Bytes decryptedKey = keyStore.decrypt(PASSWORD);
+    assertEquals(BLS_PRIVATE_KEY, decryptedKey);
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidKeyStorePaths")
+  void loadingKeyStoreWithInvalidKdfParamsThrowsException(final Path invalidJsonPath) {
+    assertThrows(JsonMappingException.class, () -> KeyStoreLoader.loadFromFile(invalidJsonPath));
+  }
+
+  @ParameterizedTest
+  @MethodSource("kdfParams")
+  void encryptWithKdfAndCipherFunction(
+      final KdfParam kdfParam, final Bytes expectedChecksum, final Bytes encryptedCipherMessage) {
```

Comment:
This probably does make sense being parameterised as we're getting multiple tests out of the same set of parameters.